django/contrib/postgres/search.py

```diff
@@ -300,3 +303,90 @@ class TrigramSimilarity(TrigramBase):
 class TrigramDistance(TrigramBase):
     function = ''
     arg_joiner = ' <-> '
+
+
+class LexemeCombinable(Expression):
+    BITAND = '&'
+    BITOR = '|'
+
+    def _combine(self, other, connector, reversed, node=None):
+        if not isinstance(other, LexemeCombinable):
+            raise TypeError(
+                'Lexeme can only be combined with other Lexemes, '
+                'got {}.'.format(type(other))
+            )
+        if reversed:
+            return CombinedLexeme(other, connector, self)
+        return CombinedLexeme(self, connector, other)
+
+    # On Combinable, these are not implemented to reduce confusion with Q. In
+    # this case we are actually (ab)using them to do logical combination so
+    # it's consistent with other usage in Django.
+    def bitand(self, other):
+        return self._combine(other, self.BITAND, False)
+
+    def bitor(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __or__(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __and__(self, other):
+        return self._combine(other, self.BITAND, False)
+
+
+class Lexeme(LexemeCombinable, Value):
+    _output_field = SearchQueryField()
+
+    def __init__(self, value, output_field=None, *, invert=False, prefix=False, weight=None):
+        self.prefix = prefix
+        self.invert = invert
+        self.weight = weight
```

Comment:
As far as I'm aware `weight` and `prefix ` are mutually exclusive, we should raise a `ValueError` in such case:
```python
raise ValueError('prefix and weight are mutually exclusive')
```

Reply:
https://www.postgresql.org/docs/10/textsearch-controls.html under section "12.3.2. Parsing Queries" there are some examples that appear to combine weights and prefix queries (and there's currently a test for this in ``postgres_tests.test_search.TestLexemes.test_as_sql``